[BUG] AssertionError: test_treespec.py::test_treespec_pickle_missing_registration

### Required prerequisites

- [x] I have read the documentation <https://optree.readthedocs.io>.
- [x] I have searched the [Issue Tracker](https://github.com/metaopt/optree/issues) that this hasn't already been reported. (comment there if it has.)

### What version of OpTree are you using?

0.14.1

### System information

`osx-arm64`:
3.12.9 | packaged by Anaconda, Inc. | (main, Feb  6 2025, 12:55:12) [Clang 14.0.6 ] darwin

### Problem description

The test `test_treespec_pickle_missing_registration` fails on `osx-arm64`

### Reproducible example code

https://github.com/metaopt/optree/blob/v0.14.1/tests/test_treespec.py#L459-L515

### Traceback

```pytb
=================================== FAILURES ===================================
__________________ test_treespec_pickle_missing_registration ___________________

    def test_treespec_pickle_missing_registration():
        if (
            sys.version_info[:2] == (3, 11)
            and platform.system() == 'Windows'
            and sysconfig.get_config_vars().get('EXT_SUFFIX', '').startswith('_d')
        ):
            pytest.skip('Python 3.11 on Windows has a bug during PyStructSequence type deallocation.')
    
        optree.register_pytree_node(
            Foo,
            lambda foo: ((foo.x, foo.y), None),
            lambda _, children: Foo(*children),
            namespace='foo',
        )
    
        treespec = optree.tree_structure(Foo(0, 1), namespace='foo')
        serialized = pickle.dumps(treespec)
    
        try:
            output = subprocess.run(
                [
                    sys.executable,
                    '-c',
                    textwrap.dedent(
                        f"""
                        import pickle
                        import sys
    
                        try:
                            treespec = pickle.loads({serialized!r})
                        except Exception as ex:
                            print(ex)
                        else:
                            print('No exception was raised.', file=sys.stderr)
                            sys.exit(1)
                        """,
                    ).strip(),
                ],
                capture_output=True,
                check=True,
                text=True,
            )
            message = output.stdout.strip()
         except subprocess.CalledProcessError as ex:
             raise RuntimeError(ex.stderr) from ex
    
>       assert re.match(
            r"^Unknown custom type in pickled PyTreeSpec: <class '.*'> in namespace 'foo'\.$",
            string=message,
        )
E       assert None
E        +  where None = <function match at 0x1057e5580>("^Unknown custom type in pickled PyTreeSpec: <class '.*'> in namespace 'foo'\\.$", string="No module named 'test_treespec'")
E        +    where <function match at 0x1057e5580> = re.match

gh/tests/test_treespec.py:505: AssertionError
```

### Expected behavior

Test must pass.

### Additional context

_No response_

--- a/tests/helpers.py
+++ b/tests/helpers.py
@@ -23,12 +23,16 @@ import sys
 import sysconfig
 import time
 from collections import OrderedDict, UserDict, defaultdict, deque, namedtuple
+from pathlib import Path
 from typing import Any, NamedTuple
 
 import pytest
 
 import optree
 from optree.registry import __GLOBAL_NAMESPACE as GLOBAL_NAMESPACE
+
+
+TEST_ROOT = Path(__file__).absolute().parent
 
 
 PYPY = platform.python_implementation() == 'PyPy'

--- a/tests/test_treespec.py
+++ b/tests/test_treespec.py
@@ -17,6 +17,7 @@
 
 import contextlib
 import itertools
+import os
 import pickle
 import platform
 import re
@@ -35,6 +36,7 @@ from helpers import (
     GLOBAL_NAMESPACE,
     NAMESPACED_TREE,
     PYPY,
+    TEST_ROOT,
     TREE_STRINGS,
     TREES,
     MyAnotherDict,
@@ -474,6 +476,7 @@ def test_treespec_pickle_missing_registration():
     treespec = optree.tree_structure(Foo(0, 1), namespace='foo')
     serialized = pickle.dumps(treespec)
 
+    python_path = os.pathsep.join([str(TEST_ROOT), os.getenv('PYTHONPATH', '.')])
     try:
         output = subprocess.run(
             [
@@ -494,6 +497,7 @@ def test_treespec_pickle_missing_registration():
                     """,
                 ).strip(),
             ],
+            env={'PYTHONPATH': python_path},
             capture_output=True,
             check=True,
             text=True,

--- a/tests/test_treespec.py
+++ b/tests/test_treespec.py
@@ -17,7 +17,6 @@
 
 import contextlib
 import itertools
-import os
 import pickle
 import platform
 import re
@@ -476,7 +475,6 @@ def test_treespec_pickle_missing_registration():
     treespec = optree.tree_structure(Foo(0, 1), namespace='foo')
     serialized = pickle.dumps(treespec)
 
-    python_path = os.pathsep.join([str(TEST_ROOT), os.getenv('PYTHONPATH', '.')])
     try:
         output = subprocess.run(
             [
@@ -486,6 +484,8 @@ def test_treespec_pickle_missing_registration():
                     f"""
                     import pickle
                     import sys
+
+                    sys.path.insert(0, {str(TEST_ROOT)!r})
 
                     try:
                         treespec = pickle.loads({serialized!r})
@@ -497,7 +497,6 @@ def test_treespec_pickle_missing_registration():
                     """,
                 ).strip(),
             ],
-            env={'PYTHONPATH': python_path},
             capture_output=True,
             check=True,
             text=True,

